From ea59830ec358f284e12bd5f109a89bf65a656763 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:39 -0400
Subject: [usb] usbtest: add a test case to support bos for queue control

Message-id: <1410887463-20674-208-git-send-email-dzickus@redhat.com>
Patchwork-id: 95518
O-Subject: [RHEL7 PATCH 207/591] usb: usbtest: add a test case to support bos for queue control
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit c952a8ba7136505cd1ca01735cc748ddc08c7d2f
Author: Huang Rui <ray.huang@amd.com>
Date:   Mon Nov 4 21:11:53 2013 +0800

    usb: usbtest: add a test case to support bos for queue control

    In Test 10 of usbtest module, it queues multiple control messages and
    thereby tests control message queuing, protocol stalls, short reads, and
    fault handling. And this patch add a test case to support queue BOS control
    request for USB 3.0 SPEC.

    Signed-off-by: Huang Rui <ray.huang@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index b415282..f3c3136 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -944,7 +944,7 @@ struct ctrl_ctx {
  int   last;
 };
 
-#define NUM_SUBCASES 15  /* how many test subcases here? */
+#define NUM_SUBCASES 16  /* how many test subcases here? */
 
 struct subcase {
  struct usb_ctrlrequest setup;
@@ -1218,6 +1218,15 @@ test_ctrl_queue(struct usbtest_dev *dev, struct usbtest_param *param)
    }
    expected = -EREMOTEIO;
    break;
+  case 15:
+   req.wValue = cpu_to_le16(USB_DT_BOS << 8);
+   if (udev->bos)
+    len = le16_to_cpu(udev->bos->desc->wTotalLength);
+   else
+    len = sizeof(struct usb_bos_descriptor);
+   if (udev->speed != USB_SPEED_SUPER)
+    expected = -EPIPE;
+   break;
   default:
    ERROR(dev, "bogus number of ctrl queue testcases!\n");
    context.status = -EINVAL;
-- 
1.7.1